app/controllers/resources_controller.rb

```diff
@@ -127,26 +127,43 @@ def file_identifier(object_id, resource_id)
     "#{ID_NAMESPACE}/file/#{object_id.delete_prefix('druid:')}-#{resource_id}/#{SecureRandom.uuid}"
   end
 
-  # rubocop:disable Metrics/AbcSize
-  def decorate_file(file:, version:, external_id: nil)
-    blob = blob_for_signed_id(file.delete(:externalIdentifier), file[:filename])
-    file[:externalIdentifier] = external_id if external_id
-    file[:version] = version
+  def metadata_for_blob(blob, file)
+    file.delete(:externalIdentifier)
     file[:size] = blob.byte_size
     file[:hasMimeType] = blob.content_type || 'application/octet-stream'
     declared_md5 = file[:hasMessageDigests].find { |digest| digest.fetch(:type) == 'md5' }.fetch(:digest)
     calculated_md5 = base64_to_hexdigest(blob.checksum)
     raise BlobError, "MD5 mismatch for #{file[:filename]}" if declared_md5 != calculated_md5
   end
-  # rubocop:enable Metrics/AbcSize
+
+  def metadata_for_file(globus_file, file)
+    return unless File.exist?(globus_file)
+
+    file[:size] = File.size(globus_file)
```

Comment:
Don't we also need to add message digests and mime type?

Reply:
I'm not sure we have that.

Reply:
Maybe? tests don't fail and I'm pretty sure we only had them if it was ActiveStorage and not a file on disk before. I can work through it.

Reply:
@justinlittman Aaron and I are going to deploy this to QA tomorrow and do some testing to see what's needed. Thanks for the heads-up.